### DISCUSSION TITLE
are StorageMetrics of Folders retrieved using CSOM or Graph?

### DISCUSSION THREAD

**Discussion by MSFT-klpinhac** [pnp/pnpcore#1585] [Q&A] [answered]
The PnP-PowerShell cmdlet 'Get-PnPFolderStorageMetric' is using the following code:  
( please see https://github.com/pnp/powershell/blob/dev/src/Commands/Files/GetFolderStorageMetric.cs )
```
IFolder folderWithStorageMetrics = Connection.PnPContext.Web.GetFolderByServerRelativeUrl(targetFolder.ServerRelativeUrl, f => f.StorageMetrics);
var storageMetrics = folderWithStorageMetrics.StorageMetrics;
```
and i know that pnpcore is using either the SharePoint REST API or the Graph API - so which one is used here?
Do settings like  "PnPContext" / "GraphFirst" make a difference here?

**Comment by jansenbe** [comment, marked as answer]
@MSFT-klpinhac : StorageMetrics are collected using a REST call, the mentioned setting is not influencing that.
